### PR TITLE
search by id

### PR DESCRIPTION
**Pull request by Hartooro**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -57,4 +57,22 @@ describe('2. GET API ARTICLES', () => {
...
     });
-});
\ No newline at end of file
+});
+describe('4 get/api/articles/:article_id', () => {
+    test('code 200, returns article with specific ID', () => {
+        return request(app).get("/api/articles/1").expect(200).then(({body})=> {
+           expect(body.article.article_id).toBe(1)
+        }
+
```
Comment (xViolaine) [inline __tests__/app.test.js]: This test needs to be more extensive.
You are correct about checking for the specific article_id, so good job here, but you should still use expect.objectContaining or .toMatchObject as well as just checking the other properties again (with expect.any(Number/String/etc.)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -57,4 +57,22 @@ describe('2. GET API ARTICLES', () => {
...
+           expect(body.article.article_id).toBe(1)
+        }
+        
+        )
+    });
+    
+  test('when searching an id not present, returns empty array', () => {
+    return request(app).get('/api/articles/122').expect(200).then((res)=>{
+      expect(res.body).toEqual([])
+    });
```
Comment (xViolaine) [inline __tests__/app.test.js]: If an ID is not present, we don't want to return an empty array, we want to return a 404 + Error, not a 200

File: models.js [hunk 3]
```diff
@@ -19,4 +19,26 @@ exports.selectArticles = () => {
...
+          })
+     }
+
+     return db
+     .query(`SELECT * FROM articles 
+      WHERE article_id = $1;
+      `,[id]).then((result)=>{
+        if (result.rows.length === 0){
+            return []
+        } else {
```
Comment (xViolaine) [inline models.js]: If the ID that is put in is a number, but the article doesn't exist, then we want to return a 404, not an empty array

File: controllers.js [hunk 4]
```diff
@@ -10,7 +10,23 @@ exports.getTopics = (req,res,next) => {
...
     .catch(next)
 }
+
+exports.getArticleById = (req, res,next) => {
+    const {id}= req.params;
+    fetchArticleById(id)
+    .then((result)=>{
+        if (result.length === 0){
+            
+            res.status(200).send(result)
```
Comment (xViolaine) [inline controllers.js]: see comment below

**Review by xViolaine — CHANGES_REQUESTED** [review]

**Review by Rayhaan2k9 — APPROVED** [review]
Nice work! 👏

**Pull request merged**